client/src/Client.js

```diff
@@ -0,0 +1,145 @@
+import axios from "axios";
+
+function checkStatus(response) {
+  if (response.status >= 200 && response.status < 300) {
+    return response;
+  } else {
+    const error = new Error(`${response.statusText}`);
+    error.status = response.statusText;
+    error.response = response;
+    throw error;
+  }
+}
+
+function parseData(response) {
+  return response.data;
+}
+
+export class ArticlesClient {
+  getArticle(data) {
+    return axios
+      .get("http://localhost:3001/api/articles", {
+        headers: {
+          Accept: "application/json",
+        },
+      })
+      .then(checkStatus)
+      .then(parseData)
+      .then((res) => res.find((article) => article.id === data.articleId));
+  }
+
+  getArticles() {
+    return axios
+      .get("/api/articles", {
+        headers: {
+          Accept: "application/json",
+        },
+      })
+      .then(checkStatus)
+      .then(parseData);
+  }
+
+  getUserArticles(data) {
+    return axios
+      .get("/api/articles", {
+        headers: {
+          Accept: "application/json",
+        },
+      })
+      .then(checkStatus)
+      .then(parseData)
+      .then((res) => res.filter((article) => article.userId === data.userId));
```

Comment:
This should be handled in the back-end not fetch all articles and then filter them in the front-end. This is not the best use of bandwidth and may cause latency

Reply:
Fixed